api/instancemanager/v1/types.go

```diff
@@ -0,0 +1,106 @@
+package v1
+
+type InsertHostRequest struct {
+	// [REQUIRED]
+	CVDInfo *CVDInfo `json:"cvd_info"`
+	// [REQUIRED]
+	HostInfo *HostInfo `json:"host_info"`
+}
+
+type CVDInfo struct {
+	// [REQUIRED] A number that uniquely identifies the a set of builds of different targets.
+	BuildID string `json:"build_id"`
+	// [REQUIRED] A string to determine the specific product and flavor from the set of builds, e.g. aosp_cf_x86_64_phone-userdebug
+	Target string
+	// The number of CVDs to create. Use this field if creating more than one instance.
+	InstancesNumber int `json:"instances_number"`
+}
+
+type HostInfo struct {
+	// Required if using GCP.
+	GCP *GCPInstance `json:"gcp"`
+}
+
+type GCPInstance struct {
+	// [REQUIRED]
+	DiskSizeGB int `json:"disk_size_gb"`
+	// [REQUIRED] More info about this field here for https://cloud.google.com/compute/docs/reference/rest/v1/instances/insert#request-body
+	MachineType string `json:"machine_type"`
+	// More info about this field here for https://cloud.google.com/compute/docs/reference/rest/v1/instances/insert#request-body
+	MinCPUPlatform string `json:"min_cpu_platform"`
+}
+
+type Operation struct {
+	Name string
+	// Service-specific metadata associated with the operation.  It typically
+	// contains progress information and common metadata such as create time.
+	Metadata interface{} `json:",omitempty"`
+	// If the value is `false`, it means the operation is still in progress.
+	// If `true`, the operation is completed, and either `error` or `response` is
+	// available.
+	Done bool
```

Comment:
it avoids enforcing to populate the underlying result object. some api might chose to populate the underlying result object whereas others not.

Reply:
One way forces to populate a result field, but it can populate it with just {} since everything inside it is omitempty.
The other way forces to "populate" a done field with either 'true' or 'false'.
Both approaches have the same number of "forces to populate". The only real difference I see is that the first one forces the client to check the result field is populated after it already checked done is 'true'.

Reply:
discussed offline and decided to keep the `done` field due clarity of use